linked_list/single_linked_list.c

```diff
@@ -0,0 +1,311 @@
+#include <stdio.h>
+#include <stdlib.h>
+#define ASSERT(x) printf("Assertion: " #x " = %d\n", (x))
```

Comment:
As the assert is a macro, make it print the current function name (`__func__`) and line number (`__LINE__`) (without it, it could have been a function).

Also, make it print only when the condition is not met, so maybe:
```C
if (!x) printf(...)
```